app/controllers/vertices_controller.rb

```diff
@@ -75,4 +76,10 @@ def set_hide_array
                      .select { |h| @params_v[h] == '1' }
                      .map { |h| h.remove('hide-').to_h }
   end
+
+  def check_permission
+    return if current_user.admin
+    return if current_user.can_edit?(@quiz)
+    redirect_to :root, alert: I18n.t('controllers.unauthorized')
+  end
```

Comment:
this also seems to be a task for cancancan

Reply:
Indeed. It has been moved there.

Reply:
still there?